runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java

```diff
@@ -109,12 +122,114 @@ public void onMenuOptionClicked(MenuOptionClicked event)
 		if (event.getMenuOption().contains("Clear history"))
 		{
 			messageQueue.removeIf(e -> e.getType() == ChatMessageType.PRIVATE_MESSAGE_RECEIVED ||
-					e.getType() == ChatMessageType.PRIVATE_MESSAGE_SENT || e.getType() == ChatMessageType.PRIVATE_MESSAGE_RECEIVED_MOD);
+				e.getType() == ChatMessageType.PRIVATE_MESSAGE_SENT || e.getType() == ChatMessageType.PRIVATE_MESSAGE_RECEIVED_MOD);
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("chathistory"))
+		{
+			populateAllowedMessages();
+
+			// If the event's new value is false, it means we need to check the message queue
+			// for any unwanted message types and remove them before the next hop or login
+			if (event.getNewValue().equals("false"))
+				purgeUnwantedMessages(event.getKey());
```

Comment:
I would rather just not replay messages based on config than try to filter the message queue based on the config.